java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -327,6 +329,40 @@ void testReadJSONFile() {
     }
   }
 
```

Comment:
Here is a new test that we can add so that we cover the case when the `MixedTypesAsStrings` feature is disabled.

```suggestion
  @Test
  void testReadMixedType2JSONFileFeatureDisabled() {
    Schema schema = Schema.builder()
            .column(DType.STRING, "a")
            .build();
    JSONOptions opts = JSONOptions.builder()
            .withLines(true)
            .withMixedTypesAsStrings(false)
            .build();
    try (Table expected = new Table.TestBuilder()
            .column("[1,2,3]", "{ \"b\": 1 }" )
            .build()) {
      assertThrows(CudfException.class, () ->
        Table.readJSON(schema, opts, TEST_MIXED_TYPE_2_JSON));
    }
  }
```